### PR TITLE
Scene tree

### PR DESCRIPTION
**Pull request by danschultz**
## What

Adds an initial implementation of the scene tree component. Should be in a usable state to pull into an application. Currently supports:

* Wiring to a viewer to sync tokens / scene view.
* Paging data with virtualized list
* Expand and collapse all.

## Ticket

https://vertexvis.atlassian.net/browse/SDK-1247
https://vertexvis.atlassian.net/browse/SDK-1452

### CODE REVIEW

File: packages/viewer/src/components/scene-tree/readme.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+| Name                       | Description                                |
+| -------------------------- | ------------------------------------------ |
+| `--scene-tree-row-height`  | The height of each row in the scene tree.  |
+| `--scene-tree-row-padding` | The padding of each row in the scene tree. |
+
+
+----------------------------------------------
+
+*Built with [StencilJS](https://stenciljs.com/)*
```
[the remark this answers is not in the document]
Comment (danschultz) [reply packages/viewer/src/components/scene-tree/readme.md:76]: Right now, the scene-tree requires the viewer, as the viewer is the thing that issues tokens. Beyond that, I'd like to keep the components within the same package. StencilJS contains polyfills and other helpers that would be optimized if we keep these components in a single package. This package probably deserves a rename. I'm thinking `@vertexvis/components`.

File: packages/viewer/src/components/scene-tree/scene-tree.tsx [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+  @Method()
+  public async scrollToIndex(index: number): Promise<void> {
+    // TODO(dan): Add alignment to top, center, or bottom.
+    const i = Math.max(0, Math.min(index, this.totalRows));
+
+    if (this.computedItemHeight != null) {
+      const top = i * this.computedItemHeight;
+      this.el.scrollTo({ top, behavior: 'smooth' });
+    }
+  }
```
[the remark this answers is not in the document]
Comment (danschultz) [reply packages/viewer/src/components/scene-tree/scene-tree.tsx:229]: Features like visibility and auto-scroll still need to be added. I'll create some additional tickets to capture this work.

[the remark this answers is not in the document]
Comment (danconner-vertex) [reply packages/viewer/src/components/scene-tree/scene-tree.tsx:229]: 👍

File: packages/viewer/src/components/scene-tree/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (danconner-vertex) [reply packages/viewer/src/components/scene-tree/readme.md:76]: 👍

**Review by danconner-vertex — APPROVED** [review]

**Review by therockstorm — APPROVED** [review]

**Pull request merged**